Scripts/RemediationScripts/Remediate-EnableHTTPSUrlSchemeForAPIManagementServices.ps1

```diff
@@ -0,0 +1,769 @@
+        }
+
+        Write-Host $([Constants]::DoubleDashLine)
+        Write-Host "[Step 4 of 4] Enabling HTTPS url scheme for API(s) of API Management services..." -ForegroundColor $([Constants]::MessageType.Warning)
+
+        # To hold results from the remediation.
+        $apiManagementServicesRemediated = @()
+        $apiManagementServicesSkipped = @()
+
+        $apiManagementWithoutHttpsEnabled | ForEach-Object {
+            $apiManagementservice = $_
+            $resourceGroupName = $_.ResourceGroupName
+           
+            Write-Host "Enabling HTTPS url scheme for API(s) of API Management Service: Resource ID - $($_.ResourceId), Resource Group Name - $($_.ResourceGroupName), Resource Name - $($_.ResourceName)" -ForegroundColor $([Constants]::MessageType.Warning)
+            
+            try
+            {   #To hold name of API(s) which are remediated and skipped.        
+                $listOfApiRemediated = @()
+                $listOfApiSkipped = @()
+                
+                $apiMgmtContext = New-AzAPIManagementContext -ResourceGroupName $_.ResourceGroupName -ServiceName $_.ResourceName
+                $apiMgmt = Get-AzAPIManagementAPI -Context $apiMgmtContext
+                $apiManagementservice.ListOfAPIWithDisabledHttps = $apiManagementservice.ListOfAPIWithDisabledHttps -split ","
+                $apiMgmt | ForEach-Object { 
+                    if($apiManagementservice.ListOfAPIWithDisabledHttps -contains  $_.Name)
+                    {
+                        Set-AzAPIManagementAPI -Context $apiMgmtContext -ApiId $_.ApiId -Protocols @("Https") -ErrorAction SilentlyContinue
+                        $output = (Get-AzAPIManagementAPI -Context $apiMgmtContext -ApiId $_.ApiId  -ErrorAction SilentlyContinue).Protocols 
+                   
+                        if($output -eq "Https")
+                        {
+                       
+                            $listOfApiRemediated += $_.Name
+                        }
+                        else
+                        {
+                            $listOfApiSkipped += $_.Name
+                        }
+                    }
+                }  
+
+                if($listOfApiRemediated.count -ne '0')
+                {
+                    $listOfApiRemediated = $listOfApiRemediated -join ","
+                    $apiManagementServicesRemediated += $apiManagementservice |Select-Object @{N='ResourceID';E={$_.ResourceId}},
+                                                                                            @{N='ResourceGroupName';E={$_.ResourceGroupName}},
+                                                                                            @{N='ResourceName';E={$_.ResourceName}},
+                                                                                            @{N='ListOfAPIRemediated';E={$listOfApiRemediated}}
```

Comment:
`ListOfAPIRemediated` to be replaced with `ListOfAPIsRemediated`. This needs to be done below as well - in the below `if` and `catch` blocks.